### PR TITLE
Remove/replace deprecated code

### PR DESCRIPTION
**Pull request by elpoelma**
This PR is a start to removing/replacing deprecated code. Specifically, it replaces the following methods:
- `toArray` is replaced by `slice`
- `firstObject` is replaced by index accessing

Additionally this PR adds the following things:
- Explicit `async: true` attributes to ember-data relationships
- Explicit `inverse` attributes to ember-data relationships if missing
- An explicit application adapter

### CODE REVIEW

**Review by lagartoverde — APPROVED** [review]

**Review by x-m-el — COMMENTED** [review]
I checked for any typos in the editing (found one).
Also a small note.

I didn't test it as it was already reviewed, so I focused on what might have gotten looked over.

File: app/models/mapping.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ export default class MappingModel extends Model {
   @attr('string') variable;
   @attr('string') type;
   @attr uri;
-  @belongsTo('code-list', { inverse: 'mappings' }) codeList;
-  @belongsTo('shape', { polyMorphic: true }) expects;
+  @belongsTo('code-list', { inverse: 'mappings', async: true }) codeList;
+  @belongsTo('shape', { inverse: null, polyMorphic: true, async: true })
+  expects;
```
[the remark this answers is not in the document]
Comment (elpoelma) [reply app/models/mapping.js:9]: @lagartoverde the `shape` model seems to be no longer used. Can you check?

[the remark this answers is not in the document]
Comment (abeforgit) [reply app/models/mapping.js:9]: > note that not specifying the inverse allows _anything_ to be set as the shape (any record of any type).

wait really? why? doesn't ED check the type of the record when you set the relationship?

[the remark this answers is not in the document]
Comment (x-m-el) [reply app/models/mapping.js:9]: It is polymorphic. So ED can't know for sure what the type should be of the relationship

**Pull request merged**